### DISCUSSION TITLE
How Are The URL and Maintainers Documentation Parsed?

### DISCUSSION THREAD

**Discussion by jgarte** [nix-community/nixvim#2572] [Q&A]
Hi,

How Are The URL and Maintainers Documentation Parsed?

I'd like to fix some that are not being parsed. Is this from the nixpkgs side? 🦆 

![image](https://github.com/user-attachments/assets/3a1d919e-651d-4a19-a403-e0a84d675a6c)

**Comment by traxys** [comment]
Do you have an example of a missing one? They are parsed through the `meta.nixvimInfo` and `meta.maintainers` nix options, most of the plugins using `mkNeovimPlugin` are automatically populated, but some plugins are older and don't use those helpers

**Reply by jgarte** [reply]
Here are a few:

https://nix-community.github.io/nixvim/plugins/alpha/index.html
https://nix-community.github.io/nixvim/plugins/cursorline/index.html
https://nix-community.github.io/nixvim/plugins/autoclose/options.html
https://nix-community.github.io/nixvim/plugins/chadtree/index.html
https://nix-community.github.io/nixvim/plugins/clangd-extensions/index.html
https://nix-community.github.io/nixvim/plugins/clipboard-image/index.html

There are probably more that I haven't listed here.

**Reply by khaneliman** [reply]
These will be missing that information because they haven't been migrated:

 list-plugins -s old
| Typ | Sty | DW | path
|-----|-----|----|--------------------------------------------------------
|    | ❌  |    | plugins/by-name/alpha/default.nix
|    | ❌  |    | plugins/by-name/autoclose/default.nix
|    | ❌  | ⚠  | plugins/by-name/clangd-extensions/default.nix
|    | ❌  |    | plugins/by-name/clipboard-image/default.nix
|    | ❌  |    | plugins/by-name/copilot-lua/default.nix
|    | ❌  |    | plugins/by-name/coverage/default.nix
|    | ❌  |    | plugins/by-name/cursorline/default.nix
|    | ❌  |    | plugins/by-name/dap/dap-go.nix
|    | ❌  |    | plugins/by-name/dap/dap-python.nix
|    | ❌  |    | plugins/by-name/dap/dap-ui.nix
|    | ❌  |    | plugins/by-name/dap/dap-virtual-text.nix
|    | ❌  |    | plugins/by-name/diffview/default.nix
|    | ❌  | ⚠  | plugins/by-name/fidget/default.nix
|    | ❌  |    | plugins/by-name/git-worktree/default.nix
|    | ❌  |    | plugins/by-name/gitlinker/default.nix
|    | ❌  |    | plugins/by-name/harpoon/default.nix
|    | ❌  |    | plugins/by-name/illuminate/default.nix
|    | ❌  |    | plugins/by-name/image/default.nix
|    | ❌  |    | plugins/by-name/inc-rename/default.nix
|    | ❌  |    | plugins/by-name/lastplace/default.nix
|    | ❌  |    | plugins/by-name/lean/default.nix
|    | ❌  | ⚠  | plugins/by-name/lspsaga/default.nix
|    | ❌  | ⚠  | plugins/by-name/luasnip/default.nix
|    | ❌  |    | plugins/by-name/mark-radar/default.nix
|    | ❌  |    | plugins/by-name/marks/default.nix
|    | ❌  |    | plugins/by-name/mkdnflow/default.nix
|    | ❌  |    | plugins/by-name/multicursors/default.nix
|    | ❌  |    | plugins/by-name/navbuddy/default.nix
|    | ❌  | ⚠  | plugins/by-name/neogen/default.nix
|    | ❌  |    | plugins/by-name/neorg/default.nix
|    | ❌  |    | plugins/by-name/notify/default.nix
|    | ❌  |    | plugins/by-name/nvim-bqf/default.nix
|    | ❌  |    | plugins/by-name/nvim-colorizer/default.nix
|    | ❌  |    | plugins/by-name/nvim-jdtls/default.nix
|    | ❌  |    | plugins/by-name/nvim-osc52/default.nix
|    | ❌  | ⚠  | plugins/by-name/nvim-tree/default.nix
|    | ❌  |    | plugins/by-name/obsidian/options.nix
|    | ❌  |    | plugins/by-name/ollama/default.nix
|    | ❌  |    | plugins/by-name/persistence/default.nix
|    | ❌  |    | plugins/by-name/presence-nvim/default.nix
|    | ❌  |    | plugins/by-name/rust-tools/default.nix
|    | ❌  |    | plugins/by-name/spider/default.nix
|    | ❌  |    | plugins/by-name/startup/default.nix
|    | ❌  |    | plugins/by-name/ts-context-commentstring/default.nix
|    | ❌  | ⚠  | plugins/by-name/wilder/default.nix
|    | ❌  |    | plugins/by-name/wtf/default.nix
|    | ❌  |    | plugins/cmp/options/default.nix
|    | ❌  |    | plugins/cmp/sources/cmp-ai.nix
|    | ❌  |    | plugins/cmp/sources/cmp-git.nix
|    | ❌  | ⚠  | plugins/cmp/sources/cmp-tabby.nix
|    | ❌  |    | plugins/cmp/sources/cmp-tabnine.nix
|    | ❌  |    | plugins/cmp/sources/copilot-cmp.nix
|    | ❌  |    | plugins/cmp/sources/crates-nvim.nix
|    | ❌  |    | plugins/lsp/default.nix
|    | ❌  |    | plugins/lsp/language-servers/efmls-configs.nix
|    | ❌  |    | plugins/lsp/language-servers/pylsp.nix
|    | ❌  |    | plugins/pluginmanagers/lazy.nix
|    | ❌  |    | plugins/pluginmanagers/packer.nix

**Reply by jgarte** [reply]
TIL 🦆

**Reply by traxys** [reply]
You can still add the URL to the documentation by specifying the `meta.nixvimInfo` value we do that in `plugins/lsp/language-servers/_mk-lsp.nix` for example:
```nix
  meta.nixvimInfo = {
    # TODO: description
    url = args.url or opt.package.default.meta.homepage or null;
    path = [
      "plugins"
      "lsp"
      "servers"
      name
    ];
  };
```

**Reply by jgarte** [reply]
TIL

**Reply by jgarte** [reply]
@traxys Should we do that until upstream resolves the propagation of that documentation?

**Reply by traxys** [reply]
I'm not sure I understand what upstream you are talking about? Those links are generated solely by nixvim, so if we want to change something about them it must be done in nixvim.
I don't think anyone would have an issue if you open a PR to add the project URL to the plugins which don't use mkNeovimPlugin
